### PR TITLE
Snake NFT

### PR DESCRIPTION
**Pull request by cmraven9**
Hello,

Just my first try at learning about the process to mint an NFT contract and adding a simple gif snake, snek NFT. I am not sure if animated gifs are allowed, so I made a png backup just in case.

UPDATED 1/06/2022: Here is the contract info https://polygonscan.com/address/0x61af7472dc596966ca694e8aef54b026cd003891

snek
snek

### CODE REVIEW

**Comment by adamhannigan** [comment]
@cmraven9 Awesome contribution, thanks!

Have you thought about how the follow up functionality would work - scare away mice and rats from crops?

**Comment by cmraven9** [comment]
@adamhannigan I am planning to create a new set of resources and its matching production chain focused on pest control or capture. There will be various animals that will "farm" the various pests or prey as a resource. For example, a sneak acquiring rats/mice with a flying hawk a superior upgraded version of the snake.  I am focused on keeping the capturing mechanic family friendly; Do you have any opinions on adding things such as hunting? 

Just a few examples of interactions:

- Mosquito control via dragonflies upgrade to bugzapper 
- Groundhogs/rabbits capture via Dog upgrade Dog2.0 
- Rats/mice capture via Snake then Hawk
- Deer capture via lures/traps

Future Framework:

- Standardize a naming convention for this set (Pest Patrol)
- Creation of resource sprites
- Creation of resource NFT
- Add an end reward for the production chain (NFT, donation, physical plushie?)
- Adding capability from other community creations (dog, cat, scarecrow)
- Use experience learned towards a major project (Electrical power systems and appliances using solar, wind, water green energy)

A fairly small project to start, but it helps me to refresh up on the coding and how to tie in NFTs (new to dapps).

**Comment by cmraven9** [comment]
snek_ingame

**Comment by adamhannigan** [comment]
@cmraven9 You will need to link the Snek smart contract up to our CommunityCraft contract.

There is a `createRecipe` on that contract that can be used to add the Snek. The QuickSwap router does not work on testnet so unfortunately you will need to test the real deal.

I would suggest using `createRecipe` with a very low price 100 gwei at the beginning while testing. When you are ready you can deploy the real contract with the real price.

File: src/dapp/types/crafting.ts [hunk 1]
```diff
@@ -476,27 +476,22 @@ export const recipes: Recipe[] = [
...
-    image: snek,
+    image: pestpsnek,
     type: "NFT",
-    address: "0x88aB216fb36A3D84Cfb67C744Fc5554F21360015",
+    address: "0x61af7472dC596966CA694E8AEF54B026cd003891",
     ingredients: [
       {
         name: "$SFF",
-        amount: 10,
+        amount: 0.0000001,
```
Comment (cmraven9) [inline src/dapp/types/crafting.ts]: testing

File: src/dapp/types/crafting.ts [hunk 2]
```diff
@@ -484,7 +484,7 @@ export const recipes: Recipe[] = [
     ingredients: [
       {
         name: "$SFF",
-        amount: 0.0000001,
+        amount: 10,
```
Comment (cmraven9) [inline src/dapp/types/crafting.ts:487]: live

**Comment by cmraven9** [comment]
@adamhannigan Thanks so much for the information, it really helps a beginner like me. Please let me know what parts need to fixed/changed.

I have been making some overall naming convention changes to my files as to not interfere with the other creators assets. 

I also believe that I got the new Snek smart contract properly setup this time around. I followed these steps:
1. Created, compiled, deployed new Snek smart contract
2. Under Deploy on the new Snek smart contract, minted 1 quantity to the same Snek smart contract address
3. Under Deploy on the new Snek smart contract, passMinterRole to CommunityCrafting contract
4. Copied, compiled existing CommunityCrafting contract
5. Using At Address for CommunityCrafting contract used createRecipe to link new Snek smart contract, SFF, and my wallet address.

New Snek smart contract address 0x61af7472dc596966ca694e8aef54b026cd003891

**Comment by Pranav-yadav** [comment]
@cmraven9 & @adamhannigan could you pls accept my friend request on discord 👉 @ 127.0.0.1
I want to discuss something related to CommunityCrafting

**Comment by adamhannigan** [comment]
Closing due to the new structure of the repo. The team is working on the new community crafting guide 👌

**Pull request closed without merging**